test/unit/reductions/grid_search/test_grid_generator.py

```diff
@@ -0,0 +1,102 @@
+
+
+@pytest.mark.parametrize("grid_limit", [0.1, 0.5, 1, 2, 3, 10, 99, 100, 101])
+def test_grid_generator_equalized_odds_basic(grid_limit):
+    # Equalized odds has four rows with potential non-zero values in the grid.
+    # grid_size = 5 ensures that each of the groups have their own column.
+    grid_size = 5
+
+    disparity_moment = EqualizedOdds()
+    label0 = 'label=0'
+    label1 = 'label=1'
+    events = [label0, label1]
+
+    grid = calculate_grid(grid_limit, grid_size, disparity_moment, events)
+
+    expected_index = pd.MultiIndex.from_product([['+', '-'], events, [0, 1]],
+                                                names=[_SIGN, _EVENT, _GROUP_ID])
+    assert (expected_index == grid.index).all()
+    expected_grid = pd.DataFrame()
+    for i in range(grid_size):
+        expected_grid[i] = pd.Series(0.0, index=expected_index)
+
+    expected_grid[0]['-', label0, 1] = grid_limit
+    expected_grid[1]['-', label1, 1] = grid_limit
+    expected_grid[3]['+', label1, 1] = grid_limit
+    expected_grid[4]['+', label0, 1] = grid_limit
+
+    assert np.isclose(expected_grid.values, grid.values).all()
+
+
+@pytest.mark.parametrize("grid_limit", [0.1, 0.5, 1, 2, 5, 10, 99, 100])
+@pytest.mark.parametrize("grid_size", [13])
+def test_grid_generator_equalized_odds(grid_limit, grid_size):
+    # Equalized odds has four rows with potential non-zero values in the grid.
+    # With grid_size = 13 we get exactly one column with the grid_limit value per row,
+    # one column with half the grid_limit value per row, and combinations of rows
+
+    disparity_moment = EqualizedOdds()
+    label0 = 'label=0'
+    label1 = 'label=1'
+    events = [label0, label1]
+
+    grid = calculate_grid(grid_limit, grid_size, disparity_moment, events)
+
+    expected_index = pd.MultiIndex.from_product([['+', '-'], events, [0, 1]],
+                                                names=[_SIGN, _EVENT, _GROUP_ID])
+    assert (expected_index == grid.index).all()
+    expected_grid = pd.DataFrame()
+    for i in range(grid_size):
+        expected_grid[i] = pd.Series(0.0, index=expected_index)
+
+    gl = grid_limit  # abbreviation for readibility
+    expected_grid.loc['+', label0, 1] = [0, 0, 0, 0, 0, 0, 0, 0, 0, gl/2, gl/2, gl/2, gl]
+    expected_grid.loc['+', label1, 1] = [0, 0, 0, gl/2, 0, 0, 0, gl/2, gl, 0, 0, gl/2, 0]
+    expected_grid.loc['-', label0, 1] = [gl, gl/2, gl/2, gl/2, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+    expected_grid.loc['-', label1, 1] = [0, gl/2, 0, 0, gl, gl/2, 0, 0, 0, gl/2, 0, 0, 0]
+
+    assert np.isclose(expected_grid.values, grid.values).all()
+
+
+def calculate_grid(grid_limit, grid_size, disparity_moment, events):
+    X, y, A = _utilities._quick_data()
+
+    disparity_moment.load_data(X, y, sensitive_features=A)
+
+    return _GridGenerator(grid_size, grid_limit,
+                          disparity_moment.pos_basis, disparity_moment.neg_basis,
+                          disparity_moment.neg_basis_present, False).grid
```

Comment:
Do we have tests with whatever that last argument is set to True?

Reply:
No, and I'm not going to add them now since I don't quite understand what scenario it's for. The test gap wasn't my main intention here. It should still be addressed at some point, though.